### ISSUE TITLE
Sprint 1.30.0 2/2 - Nov 1 to Nov 12

### ISSUE THREAD

**Issue by marcushyett-ph** [PostHog/posthog#6651]
# Global Sprint Planning

## Agenda
* Retro built into the status update:
    * Could you / we have been working on something more impactful?
    * Why did / didn’t we complete X?
    * Could we have done a bigger version of Y?
 
## Retro: Status of Outcomes from Previous Sprint

1. **Correlation Analysis (Owner: @neilkakkar) - Ship the best Quant analysis tool in the industry (50:50)** 
    - Expected End of Sprint Status: Our metric for shipping the best quant tool: Get 3 users to absolutely love correlation analysis. Today, we're at 1/3. By Friday, we hope to talk to 2 more customers whose pain points we've solved through (new) features that make correlations very compelling. We still need to polish UX around this though. Another week to close open loops would be wonderful.
2. **Data Integrity Querying (Owner: @EDsCODE) - Resolve and build robust tests for all known query inconsistency issues affecting persons modal**
    - Expected End of Sprint Status: Only had 1 week for this but will get the known issues fixed, we need to do something preventative, expect to complete it. A more impactful version would have been to analyze the root cause and prevent it.
3. **Data Integrity Ingestion**
    - "Ensuring Postgres and CH persons are fully in sync" @yakkomajuri 
        - Expected End of Sprint Status: Most impactful - problematic, leading to incorrect cohorts. We're not going to complete it because of circular dependencies, we know what to do, PRs are out and under review. We could've made sure we have all tables in sync (not viable right now - did explore it). We need to do internal planning when there are potential conflicts.
    - "Allowing events to be ingested in any order" @tiina303 
        - Expected End of Sprint Status: PostHog team will have set and set_once work in a way that is ingestible in any order, we want to roll it out slowly - the implications of getting it wrong are severe. It was most impactful, we didn't reach it because of parallelizing the work. One way this could've been bigger, a test to validate events in any order - ingest them both ways and compare the results.
4.  **Recordings (Owner: @rcmarron)**
    - 99% of session recordings have a full snapshot (and are therefore not missing)
        - Expected End of Sprint Status: We hit this goal for users that have updated their posthog-js. Over the next month or so, we will continue to see the overall metric go down as more customers upgrade.
    - 80% of the time I should be able to find the part of the session I care about within 5 seconds. Measured via internal tests
        - Expected End of Sprint Status: By the end of the sprint, we should have a player that loads much much quicker (hundreds of milliseconds instead of seconds) and displays events on seek bar to quickly find the moment you're looking for. We will be very close to meeting our goal. There will still be a bit more work to finish implementing the full player as designed (events card on the right and handling periods of inactivity)
4. **Groups (Owner: @macobo) - Full group analytics support enabled for 5 alpha customers (+ release plan)**
     - Expected End of Sprint Status: Library support + ingestion work ready. Full support behind Karl being sick

## Retro: What can we do better next sprint?
_Retro on the global sprint planning / goals? What went well? What did not go well? Were the goals too big or too small? How can we do to improve for this sprint?_ 

**Please edit this issue and add any retro items you have**

1.  CoreExp: We knew this going in but the Core Exp goal was pretty open ended and while we know we made a ton of progress it's hard to quantify with a clearer goal (hence proposal above).
2.  CoreExp: Should've had clearer priority & goals around multi-property breakdown (potentially more resources, or narrower scope). This was a blocking Enterprise feature, that we would've ideally wanted to get out the door.
3. 
4.
5.

## Plan: Proposed Goals for Next Sprint 

Each goal should have a single owner. Owner can only be an engineer. Please feel free to iterate on these goals or suggest alternatives - we'll discuss during the split planning session.

1.  **Product Hunt Launch (Owner: @mariusandra)** - Get the UX to a high quality state with new Information Architecture ready for Product Hunt Launch
    - Goal (Owner: @Twixes): Implement and ship to beta users the new IA interface.
    - Goal: Fix all UX/bugs on saved insights, Epic - Nail Seamless Analytics Experience #6513 (comment)
    - Goal: Create the "Discoveries by PH teammates" graph in under 6 seconds. Will require all the UX fixes around th taxonomic filter.
    - why? We need to be taken seriously if we're investing in a big launch
2. **Data Integrity Querying (Owner: @EDsCODE)** - Identify and resolve underlying query result issues
    - Why? This is affecting users, high impact when it shows up
3. **Groups (Owner: @macobo)** - 5 alpha users use & love groups (for insights, groups pages and feature flags)
    - Why? The second thing missing from all analytics tools, people reached out even from being a few days behind
4. **Data Integrity Ingestion (Owner: @tiina303)** - Bring ingestion in any order to everyone on cloud
    - Why? This is affecting users, high impact when it shows up
5. **Quantitative Analysis (Owner: @neilkakkar)** - Polish the experience and launch an upsell flow to drive adoption
    - Why? We need to get this out

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

## Retro

<!-- Talk about what went well, what didn't go well and any actions to improve next time -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## Planning

<!-- Each item should have a single owner. Owner can only be an engineer. -->

### High priority

-

### Low priority / side quests

-

```

**Comment by paolodamico** [comment]
**On priorities**
- From Monday's standup, I think we'll want to continue working on recordings. We still have a pretty big bug (previously unprioritized) that leads to a broken experience (overlapping CSS from multiple tabs).
- For Core Exp I think the goals/focus should be around top quality UX, rather than the Product Hunt Launch.
   - Goal proposal: Implement and ship to beta users the new IA interface.
   - Goal proposal: Fix all UX/bugs on saved insights, https://github.com/PostHog/posthog/issues/6513#issuecomment-951870899
   - Goal proposal: Create the "Discoveries by PH teammates" graph in under 6 seconds. _Will require all the UX fixes around the taxonomic filter._
- For "Data Integrity Querying"is the proposed goal around testing the best way to address this? _We already had this goal and it doesn't seem to be addressing the underlying issue. Also I'll defer to @EDsCODE, but I don't know if we have reached the conclusion that the best _

**On retrospective**
- We knew this going in but the Core Exp goal was pretty open ended and while we know we made a ton of progress it's hard to quantify with a clearer goal (hence proposal above).
- Should've had clearer priority & goals around multi-property breakdown (potentially more resources, or narrower scope). This was a blocking Enterprise feature, that we would've ideally wanted to get out the door.

**Comment by fuziontech** [comment]
Team Platform

🎧 @guidoiaquinti is support hero 🦸 

## Retro

- @fuziontech - Sprint was super thrashy for me and I did not get everything that I'd like to get done, done. A lot of progress everywhere, but nothing was wrapped up in full. Ansible, Events Schema, Plugin-server, and Altinity work is all < 100% done. Going to try to push to get some of this done before Monday.
- @yakkomajuri - Overall a good sprint _but_ also in terms of the biggest goal that was laid out. Circular dependencies will make this unshippable. Made progress in sense of delegation/sharing of work. Still figuring out how to make the process a bit better. Good progress on work that was shared with @tiina303 
- @tiina303 - We need to focus on having shippable items for each person for sprints. Circular dependencies - yakko blocked by tiina blocked by yakko <insert pointing meme>
- @guidoiaquinti - semi-good sprint. I was hoping to do more stuff but anyway. Made progress with Helm charts, PostHog security and docs. Broke production 🥳 
- @marcushyett-ph - Yes. ✈️ lag. Wants to dig into deployments to determine concretely what we want to support (and how) and what we will not support

## Hang over items from previous sprint

- @fuziontech Plugin-server monorepo (continue)
- @fuziontech Events Schema primary key rework (continue)
- @fuziontech Ansible for CH configs (continue)
- @fuziontech Altinity conference (continue)
- @guidoiaquinti - PostHog security (PROD docker image still waiting for review).
- @guidoiaquinti - Helm Charts: it’s going to take several sprints before reaching a stable state



## Planning

🎧 @guidoiaquinti is support hero 🦸 

### High priority

- @fuziontech - Deployments jam with @marcushyett-ph with a concrete plan on deployments as deliverable
- @tiina303 - set, set once, and merging on cloud ☁️  (This means out of order ingestion is shipped on ☁️ )
- @yakkomajuri - Postgres -> ClickHouse Sync
- @yakkomajuri - Ensure quality of Plugins as a product in preparation for ProductHunt launch
- @guidoiaquinti - is support hero 🦸 - Security work - Container security (Alpine base) & Helm chart, monitoring and logging
- @tiina303 - Ensure quality of Deployments (before ProductHunt launch)
   - Setup helm chart on every stack
   - Better default values for Kafka in charts (retention and size)

**Comment by mariusandra** [comment]
Team Core ~~Experience~~Aardvark

## Retro
<details><summary>Expand retro</summary>
<p>

- Alex: was kind of fun, video player logic is fairly unique, many edge cases. A lot of items, wish I had scoped down a bit. Working now on getting the API events working (extra bugs: CH can't order them, etc). Went mostly good, just the unscoped bugs took time.
- Rick: went pretty well, the two big things (full recordings, paginated decompression work). Surprise: new recordings endpoint is anyway way faster than the old one, not sure if all optimisations for pagination/decompressions might have been needed. A lot of discussion around session_id/window_id. Happy we made a decision, seems like the right one as well 👍.
- Chris: fine. Pretty reactive. Not much forward looking, but trying to get a handle and not blocking things regardless. Otherwise, it's fine. Just fine. Nothing is burning. Everything is fine.
- Michael: definitely underestimated how much effort it'd take to refactor all the logics for the project-based URL approach. Really cool that we decided to go forward with the new design. Really good progress for design, implementation started this week. Almost all of the top bar, will continue still this week. 
- Paul: (not working today). Breakdown by multiple properties, didn't get all the way, but it works now for funnels. Needs some polish and will come to the next print.
- Marius: wrote code, got turbo mode PR out, saved insights finalizing still in progress, will spill over but is the main goal for next sprint anyway.

</p>
</details>

## Hand over items from previous sprint

1. Recording playback: most likely will get events in the seekbar, but not in the sidebar. Polish + bugsquash also needed. Alex off 1 week, will work on this 2nd week.
2. Recordings: session_id/window_id work will continue. Might take the entire sprint with all the moving parts. 
3. IA: will still hang over, though looking good now.

## Planning

- Recordings  @rcmarron 
- Recording playback @alexkim205 
- IA @Twixes 
- Saved insights and all UX polish @mariusandra 
- Breakdown by multiple properties @pauldambra 
- Upsell components @clarkus 

### High priority

- See list above

### Low priority / side quests

- See lists from other teams

**Comment by EDsCODE** [comment]
## Retro

<details>
<summary>Click to expand</summary>
Eric:

- Data integrity list of issues
- Actions pie chart bug
- CSV limit bug
- Funnel argument bug
- Time to convert rendering bug
- Proposal to speed up ee tests
- Experimented with snapshot testing

Harry:

- Added property selection component, along with kea logic
- Added ability to exclude property from person property correlations
- Added ability to exclude property from event property correlations
- Wip: removing eslint react-hook-rules
- Wip: install cypress in dev
- Wip: smoothing intervals - still needs review regarding logic usage, does it need them?
- Wip: storing exclusions in team settings
- Felt like a slow two weeks, not in terms of things to do, but progress wise

Karl:

- Groups groups groups! 
- Library support + Docs live
- Li helped with Go!
- Basic ingestion live
- Schema changes (working with jams) tbd
- Posthog using groups = PR out
- Need help with the project - I am slow alone
- Helping rick, guido and yakko sporadically

Neil:

- Correlations! Lots of things here: Persons modal for everything, nested breakdowns, exclusions backend, special autocapture treatment, talking to users
- Pretty hard to switch off after work, sigh, working longer than usual
- Overestimated polishing UI time

Li:

- Support hero week 1
- Off most of week 2
- Groups work in posthog-go

</details>

## Hang over items

- Correlations UI
- Few bits of work that had been started
  - Removing eslint react-hook-rules
  - Install cypress in dev
  - Smoothing intervals - still needs review regarding logic usage, does it need them?
  - Storing exclusions in team settings

## Planning

**Data integrity: Queries** — Eric (Harry)
_Goal: Have confidence that the results returned when examining persons of an insight match the numbers counted_

- Person tests on all insight tests
- Have metrics on what filters are being used that we aren’t testing
- Clear ability to see what filter we've covered in tests
- person endpoint on insight endpoint
- parameterize tests
- Make filter typing specific to insight

_Secondary goal: Speed up backend tests by 60%+_

- Parallelizing EE tests using xdist (https://github.com/PostHog/posthog/pull/6171)
- Batching event creation (https://github.com/PostHog/posthog/pull/6725)

**Group Analytics** — Karl (Li, Neil)
_Goal: Trends and Funnel group analytics support in front of alpha customers_

Next steps for next week:

- Finish up schema changes - hopefully tonight
- BE querying - starting from trends + funnels
- FE filtering + querying experience

**Correlation Analysis Final Clean up** — Neil (Harry)

## High Priority

- Finish correlation analysis
- Moving the needle on persons related data quality
- Shipping an MVP of groups for some insights

## Low Priority

**Comment by Twixes** [comment]
Fin.
